packages/scene-composer/src/components/panels/scene-components/data-overlay/DataOverlayPanelConfigEditor.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useCallback } from 'react';
+import { Checkbox, FormField } from '@awsui/components-react';
+import { useIntl } from 'react-intl';
+
+import { IDataOverlayComponentInternal } from '../../../../store';
+import { Component } from '../../../../models/SceneModels';
+
+interface IDataOverlayPanelConfigEditorProps {
+  config: Component.OverlayPanelConfig | undefined;
+  onUpdateCallback: (componentPartial: Partial<IDataOverlayComponentInternal>, replace?: boolean | undefined) => void;
+}
+
+export const DataOverlayPanelConfigEditor: React.FC<IDataOverlayPanelConfigEditorProps> = ({
+  config,
+  onUpdateCallback,
+}) => {
+  const { formatMessage } = useIntl();
+
+  const onPinnedChange = useCallback(
+    ({ detail: { checked } }) => {
+      onUpdateCallback({ config: { isPinned: checked } }, false);
+    },
+    [onUpdateCallback],
+  );
+
+  return (
+    <FormField label={formatMessage({ defaultMessage: 'Panel Config', description: 'FormField label' })}>
+      <Checkbox data-testid='pinned-checkbox' checked={config?.isPinned === true} onChange={onPinnedChange}>
```

Comment:
I think you don't need to check explicitly here. You can just say `config.isSpinned`

Reply:
`config.isSpinned` may be undefined so I will get type error